### PR TITLE
S6 Searching

### PR DESCRIPTION
**Pull request by kuzivka**

### CODE REVIEW

File: src/components/header/SearchInput.tsx [hunk 1]
```diff
@@ -16,22 +17,41 @@ import {
...
+  const dispatch = useDispatch();
+
   const { data: categories } = useGetCategoriesQuery();
   const selectChangeHandler = useCallback((event: SelectChangeEvent) => {
-    setSelectState(event.target.value);
+    setSelectedCategory(event.target.value);
   }, []);
 
+  useEffect(() => {
+    const timer = setTimeout(() => {
```
Comment (Kost927) [inline src/components/header/SearchInput.tsx:36]: just interesting, why do you need a timer over here? Is this custom debounce?

Comment (kuzivka) [reply src/components/header/SearchInput.tsx:36]: to dispatch filter action after the user stopped typing
at first, I dispatched on every change as I did in todo application
but that is the way to optimize this

File: src/store.ts [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+  productList: productListReducer,
+});
+
 export const store = configureStore({
-  reducer: {
-    [ecommerceApi.reducerPath]: ecommerceApi.reducer,
-  },
+  reducer: rootReducer,
   middleware: (getDefaultMiddleware) =>
     getDefaultMiddleware().concat(ecommerceApi.middleware),
```
Comment (Kost927) [inline src/store.ts]: better practice to use modern syntax [...getDefaultMiddleware(), ecommerceApi.middleware], coz when you will have 10 or more middlewares you will see concat.****.concat.*****.concat

Comment (kuzivka) [reply src/store.ts]: okay

**Review by Kost927 — APPROVED** [review]

File: src/components/header/SearchInput.tsx (hunk 1 above)
Comment (Kost927) [reply src/components/header/SearchInput.tsx:36]: as I said, debounce )) You can read about throttle and debounce. You will use them often. https://codepen.io/goit-academy/pen/rNyryZP

Comment (kuzivka) [reply src/components/header/SearchInput.tsx:36]: okay)
thank you

**Pull request merged**